jolt-core/src/main/java/com/bazaarvoice/jolt/templatr/function/Math.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2013 Bazaarvoice, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bazaarvoice.jolt.templatr.function;
+
+import com.bazaarvoice.jolt.common.Optional;
+
+@SuppressWarnings( "unused" )
+public class Math {
+
+
+    public static final class MaxOf implements Function {
+        @Override
+        public Optional<Object> apply( final Object... args ) {
+            Integer maxInt = Integer.MIN_VALUE;
+            Double maxDouble = Double.MIN_VALUE;
+            Long maxLong = Long.MIN_VALUE;
+
+            if(args.length == 0) {
+                return Optional.empty();
+            }
+
+            for(Object arg: args) {
+                if(arg instanceof Integer) {
+                    maxInt = java.lang.Math.max( maxInt, (Integer) arg );
+                }
+                else if(arg instanceof Double) {
+                    maxDouble = java.lang.Math.max( maxDouble, (Double) arg );
+                }
+                else if(arg instanceof Long) {
+                    maxLong = java.lang.Math.max(maxLong, (Long) arg);
+                }
+            }
+            if(maxInt == Integer.MIN_VALUE && maxDouble == Double.MIN_VALUE && maxLong == Long.MIN_VALUE) {
```

Comment:
Cool, so to break this all I need is an input array that looks like [ -2147483648 ], and instead of getting -2147483648 I will get Optional.empty()   ;)